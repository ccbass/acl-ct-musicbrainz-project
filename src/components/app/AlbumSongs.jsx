import React, { useEffect, useState } from 'react';
import Songs from './Songs';


const AlbumSongs = () => { 
  const [recordings, setRecordings] = useState([]);
  const [loading, setLoading] = useState(true);

  const fetchRecordings = async () => { 
      
    const res = await fetch('http://musicbrainz.org/ws/2/recording?release=7d166a44-cfb5-4b08-aacb-6863bbe677d6&fmt=json');

    const recs = await res.json();

    console.log(recs);

    return recs.recordings.map(recording => ({
      title: recording.title,
      releaseDate: recording.first_release_date,
      disambiguation: recording.disambiguation,
      length: recording.length,
      id: recording.id,
      video: recording.video

    }));

  };

  useEffect(() => {
    fetchRecordings().then(setRecordings)
      .finally(() => setLoading(false));
      
  }, []);



  return (
    <div>
      <h2>Recordings</h2>
      <Songs recordings={recordings}
        loading={loading}/>
    </div>
  );
};

AlbumSongs.propTypes = { 

};

export default AlbumSongs;
