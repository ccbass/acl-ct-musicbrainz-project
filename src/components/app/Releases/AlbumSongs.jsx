import React, { useEffect, useState } from 'react';
import { useParams } from 'react-router-dom';
import Songs from './Songs';
import styles from './Releases.css';

const AlbumSongs = () => {
    const [recordings, setRecordings] = useState([]);
    const [loading, setLoading] = useState(true);
    const { release } = useParams();

    const fetchRecordings = async (release) => {
        const res = await fetch(
            `http://musicbrainz.org/ws/2/recording?release=${release}&fmt=json`
        );
        // fetch('http://musicbrainz.org/ws/2/recording?release=7d166a44-cfb5-4b08-aacb-6863bbe677d6&fmt=json');

        const recs = await res.json();

        return recs.recordings.map((recording) => ({
            title: recording.title,
            releaseDate: recording.first_release_date,
            disambiguation: recording.disambiguation,
            length: recording.length,
            id: recording.id,
            video: recording.video,
        }));
    };

    useEffect(() => {
        fetchRecordings(release)
            .then(setRecordings)
            .finally(() => setLoading(false));
    }, []);

    return (
        <div className={styles.albumdiv}>
            <div className={styles.header}>
                <h2>Recordings</h2>
            </div>
            <Songs recordings={recordings} loading={loading} />
        </div>
    );
};

export default AlbumSongs;
