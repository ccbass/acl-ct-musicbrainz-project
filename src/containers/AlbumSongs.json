{
    "recordings": [
    {
    "title": "Negative Creep",
    "first-release-date": "1989-06-01",
    "disambiguation": "",
    "length": 175693,
    "id": "06745db3-a2bc-4131-8a9d-0cbdc25b494a",
    "video": false
    },
    {
    "id": "0b3292f6-8f0d-4160-92d3-49348029a619",
    "video": false,
    "length": 162173,
    "disambiguation": "",
    "title": "School",
    "first-release-date": "1989-06-01"
    },
    {
    "disambiguation": "",
    "first-release-date": "1989-06-01",
    "title": "Love Buzz",
    "id": "1adfecb7-875f-4203-b3b1-8e2e643f94a2",
    "video": false,
    "length": 215120
    },
    {
    "length": 322240,
    "video": false,
    "id": "25bbbd76-5f48-4b37-9722-44a346c4f3a9",
    "title": "Sifting",
    "first-release-date": "1989-06-01",
    "disambiguation": "Reciprocal Recording session, 1988‐01‐23 or 1988‐06"
    },
    {
    "length": 174453,
    "id": "2837e0d2-5ce9-44da-803e-797c835d672c",
    "video": false,
    "title": "Blew",
    "first-release-date": "1989-06-01",
    "disambiguation": ""
    },
    {
    "video": false,
    "id": "4cce1ee1-38e6-4ea0-8584-b6ebeb854d7d",
    "length": 138026,
    "disambiguation": "",
    "title": "Floyd the Barber",
    "first-release-date": "1989-06-01"
    },
    {
    "length": 103800,
    "video": false,
    "id": "6011093f-704a-46b0-8941-ae2bee94d70a",
    "first-release-date": "1989-06-01",
    "title": "Downer",
    "disambiguation": ""
    },
    {
    "length": 245853,
    "video": false,
    "id": "67930d52-5e75-4c00-b36f-ee3c43bed160",
    "first-release-date": "1989-06-01",
    "title": "Paper Cuts",
    "disambiguation": "Reciprocal Recording session, 1988‐01‐23 or 1988‐06"
    },
    {
    "length": 203929,
    "id": "6cd0083c-e4ce-4745-aaf0-704ab1de988a",
    "video": false,
    "title": "Mr. Moustache",
    "first-release-date": "1989-06-01",
    "disambiguation": ""
    },
    {
    "length": 168293,
    "id": "b2e623ea-378b-4479-bb75-fa202aacbfc9",
    "video": false,
    "first-release-date": "1989-06-01",
    "title": "About a Girl",
    "disambiguation": ""
    },
    {
    "disambiguation": "",
    "first-release-date": "1989-06-01",
    "title": "Big Cheese",
    "video": false,
    "id": "c498c2ac-a571-412e-a26e-f37e86a1edda",
    "length": 222160
    },
    {
    "length": 182973,
    "video": false,
    "id": "c9972a41-2ae8-43f6-9822-ba2fec7e50d1",
    "title": "Swap Meet",
    "first-release-date": "1989-06-01",
    "disambiguation": ""
    },
    {
    "title": "Scoff",
    "first-release-date": "1989-06-01",
    "disambiguation": "",
    "length": 250106,
    "video": false,
    "id": "dbecedb1-2a90-43fc-add4-ce4016676545"
    }
    ],
    "recording-offset": 0,
    "recording-count": 13
    }