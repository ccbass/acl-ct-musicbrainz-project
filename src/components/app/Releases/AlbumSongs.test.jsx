import React from 'react';
import { fireEvent, render, screen, waitFor, cleanup } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { setupServer } from 'msw/node';
import { rest } from 'msw';
import AlbumSongs from './AlbumSongs';
import AlbumData from '../../../containers/AlbumSongs.json';

const server = setupServer(
    rest.get('http://musicbrainz.org/ws/2/recording?release=7d166a44-cfb5-4b08-aacb-6863bbe677d6&fmt=json', (req, res,ctx) => { 
        return res(ctx.json(AlbumData));
    })
);

describe('Testing for AlbumSongs Container', () => { 
    beforeAll(() => server.listen());
    afterAll(() => server.close());
    afterEach(cleanup);
    it('should render a list of album releases and route to a lyrics page', async () => { 
        render(<MemoryRouter> <AlbumSongs/> </MemoryRouter>);

        screen.getByText('Loading...');
        
        const ul = await screen.findByRole('list', { name: 'releases' });
        expect(ul).toMatchSnapshot();


    });
});
