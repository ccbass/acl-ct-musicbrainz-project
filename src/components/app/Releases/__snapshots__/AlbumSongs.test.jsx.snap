// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing for AlbumSongs Container should render a list of album releases and route to a lyrics page 1`] = `
<ul
  aria-label="releases"
  class="ul"
>
  <li
    class="list"
  >
    <a
      activeclassname="selected"
      aria-label="to-lyrics"
      class="link"
      exact="true"
      href="/lyrics/undefined/Negative Creep"
    >
      <p>
        Negative Creep
      </p>
    </a>
  </li>
  <li
    class="list"
  >
    <a
      activeclassname="selected"
      aria-label="to-lyrics"
      class="link"
      exact="true"
      href="/lyrics/undefined/School"
    >
      <p>
        School
      </p>
    </a>
  </li>
  <li
    class="list"
  >
    <a
      activeclassname="selected"
      aria-label="to-lyrics"
      class="link"
      exact="true"
      href="/lyrics/undefined/Love Buzz"
    >
      <p>
        Love Buzz
      </p>
    </a>
  </li>
  <li
    class="list"
  >
    <a
      activeclassname="selected"
      aria-label="to-lyrics"
      class="link"
      exact="true"
      href="/lyrics/undefined/Sifting"
    >
      <p>
        Sifting
      </p>
    </a>
  </li>
  <li
    class="list"
  >
    <a
      activeclassname="selected"
      aria-label="to-lyrics"
      class="link"
      exact="true"
      href="/lyrics/undefined/Blew"
    >
      <p>
        Blew
      </p>
    </a>
  </li>
  <li
    class="list"
  >
    <a
      activeclassname="selected"
      aria-label="to-lyrics"
      class="link"
      exact="true"
      href="/lyrics/undefined/Floyd the Barber"
    >
      <p>
        Floyd the Barber
      </p>
    </a>
  </li>
  <li
    class="list"
  >
    <a
      activeclassname="selected"
      aria-label="to-lyrics"
      class="link"
      exact="true"
      href="/lyrics/undefined/Downer"
    >
      <p>
        Downer
      </p>
    </a>
  </li>
  <li
    class="list"
  >
    <a
      activeclassname="selected"
      aria-label="to-lyrics"
      class="link"
      exact="true"
      href="/lyrics/undefined/Paper Cuts"
    >
      <p>
        Paper Cuts
      </p>
    </a>
  </li>
  <li
    class="list"
  >
    <a
      activeclassname="selected"
      aria-label="to-lyrics"
      class="link"
      exact="true"
      href="/lyrics/undefined/Mr. Moustache"
    >
      <p>
        Mr. Moustache
      </p>
    </a>
  </li>
  <li
    class="list"
  >
    <a
      activeclassname="selected"
      aria-label="to-lyrics"
      class="link"
      exact="true"
      href="/lyrics/undefined/About a Girl"
    >
      <p>
        About a Girl
      </p>
    </a>
  </li>
  <li
    class="list"
  >
    <a
      activeclassname="selected"
      aria-label="to-lyrics"
      class="link"
      exact="true"
      href="/lyrics/undefined/Big Cheese"
    >
      <p>
        Big Cheese
      </p>
    </a>
  </li>
  <li
    class="list"
  >
    <a
      activeclassname="selected"
      aria-label="to-lyrics"
      class="link"
      exact="true"
      href="/lyrics/undefined/Swap Meet"
    >
      <p>
        Swap Meet
      </p>
    </a>
  </li>
  <li
    class="list"
  >
    <a
      activeclassname="selected"
      aria-label="to-lyrics"
      class="link"
      exact="true"
      href="/lyrics/undefined/Scoff"
    >
      <p>
        Scoff
      </p>
    </a>
  </li>
</ul>
`;
